### PR TITLE
[test] add support for unit testing

### LINKED ISSUE #360: [testing] add support for unit tests
Implement support for building and running C++ unit tests.

### PR DESCRIPTION
**Pull request by pilkicTT**
Adding build targets for building and running unit tests. This wasn't
covered when we transitioned from previous stack.

To build/run use:

```
cmake --build build -- build_unit_tests
cmake --build build -- run_unit_tests
```

Also, docs have been updated with instructions on how to run both unit
tests and end-to-end tests.

Note: I have removed some of the old tests that were causing build
issues and for which there is no need for anymore.

Closes #360

### CODE REVIEW

**Comment by pilkicTT** [comment]
I will update the docs as well...

**Review by mtopalovicTT — APPROVED** [review]

**Comment by pilkicTT** [comment]
> I will update the docs as well...

Done...

File: docs/src/test.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+cmake --build build -- build_unit_tests
+```
+
+To run the unit tests (this will also build the tests if they are not built):
+
+```sh
+cmake --build build -- run_unit_tests
+```
+
+> **Note:** The unit tests are built in the `build/forge/csrc/tests` directory. From there, you can run targeted tests directly.
```
Comment (nvukobratTT) [inline docs/src/test.md:20]: Can you just share an example of running them directly as well?

**Review by nvukobratTT — APPROVED** [review]
LGTM! 🙌

**Pull request merged**